internal/httpsrv/server.go

```diff
@@ -214,8 +215,14 @@ func (s *Server) livenessCheck(c *gin.Context) {
 // requests. Currently our only dependency is the DB so we just ensure that it
 // is responding.
 func (s *Server) readinessCheck(c *gin.Context) {
-	if err := s.DB.PingContext(c.Request.Context()); err != nil {
-		s.Logger.Sugar().Errorf("readiness check db ping failed", "err", err)
+	startTime := time.Now()
+
+	ctx, cancel := context.WithTimeout(context.Background(), dbPingTimeout)
```

Comment:
This was my own paranoia, wanting to avoid using the existing context as I was concerned that something about it could have been involved in the errors we were seeing. You make valid points, lemme fix this.

Reply:
Ok, updated - I'm now re-using the request context, rather than a Background one.